src/workerd/jsg/util.c++

```diff
@@ -366,11 +366,28 @@ kj::String extractTunneledExceptionDescription(kj::StringPtr message) {
   return kj::str(tunneledErrorType(message).type);
 }
 
-kj::String annotateBroken(kj::StringPtr internalMessage, kj::StringPtr brokenessReason) {
+void annotateBroken(kj::Exception& exception, kj::StringPtr brokenessReason) {
   // TODO(soon) Once we support multiple brokenness reasons, we can make this much simpler.
 
+  kj::StringPtr internalMessage = exception.getDescription();
+  if (internalMessage.startsWith("broken."_kj) || internalMessage.startsWith("remote.broken."_kj)) {
+    // If we already set up a brokeness reason, we shouldn't override it.
+    return;
+  }
+
   auto tunneledInfo = tunneledErrorType(internalMessage);
 
+  if (tunneledInfo.isInternal) {
```

Comment:
I don't think this is right. Do you know the pathway where you were failing to log the internal error? Most of them end up logging internal errors so this will double log.

Reply:
Ah, I think I understand the confusion here. `tunneledInfo.type` is the entire exception message after the prefixes. It is indeed included in the returned message.

That said, I was thinking about this last night and I think a `KJ_LOG(INFO, "annotating brokenness", ...)` wouldn't hurt at all.

Reply:
> tunneledInfo.type is the entire exception message after the prefixes.

Only for tunneled types. If it's an internal error, then `tunneledInfo.type` is simply `Error`. Hence the internal error message is thrown away. Meanwhile the replacement description has the format of a tunneled JS exception, therefore is not logged downstream.

Reply:
ok, commit removed.